### DISCUSSION TITLE
ECONNREFUSED when trying to use debugpy from VSCode and implement NMCLI in docker container

### DISCUSSION THREAD

**Discussion by DLeberfinger** [microsoft/debugpy#1622] [Q&A]
I am working with some Toradex hardware and their VSCode IDE extension that implements debugpy to debug python code while running on their hardware. The setup works until I try to implement a few changes in my set up to allow the container to interface with network-manager. The team at Toradex is pointing to debugpy as being the issue area and recommended I post my issues here. 

Everything compiles and works great with debugpy until I try to add network-manager into the mix. I am able to successfully install network-manager through my docker-compose/Dockerfile. However, when I attempt to add `- /var/run/dbus:/var/run/dbus` or install network-manager it seems to break the debugpy connection, I can do both of them independently, but not at the same time. The container is successfully transferred to my board and runs, however I get connect ECONNREFUSED in VSCode when I do this.

![image](https://github.com/microsoft/debugpy/assets/88494653/ad3d7949-c775-4db0-9f0d-2ca45c200de9)

It seems like if I manually install network-manager into my container after starting it, debugpy continues to work.

I am looking for any help I can get here. Has anyone else successfully implemented NMCLI inside a docker container and debugged with debugpy? 

Are there any additional tools I can use to help understand where the issue is coming from? I tried to implement the --log-to for debugpy, but it seems like debugpy never runs so it does not create a log anywhere that I can see.

**Comment by fustfra1** [comment]
@DLeberfinger  I have the same issue. Could you solve it?

**Reply by ytwanghaoyu** [reply]
Same issue here, I think that might related to vscode, it try to auto-attach the debugger from a specific version.
